Add documentation for compact preset

tomivirkki opened the pull request:
Needs vaadin-lumo-styles^1.4

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/flow-and-components-documentation/442)
<!-- Reviewable:end -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/flow-and-components-documentation?pullRequest=442) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/flow-and-components-documentation?pullRequest=442) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/flow-and-components-documentation?pullRequest=442) it.</sub>

tomivirkki commented:
___
*[documentation/theme/using-component-themes.asciidoc, line 201 at r1](https://reviewable.io/reviews/vaadin/flow-and-components-documentation/442#-LWycqIu2me3-hkuPLct:-LWyrk1Z7CKCril-Ss9S:b-yco55a) ([raw file](https://github.com/vaadin/flow-and-components-documentation/blob/8ad19ae81d04ff643ac180eebc11198b43ebd879/documentation/theme/using-component-themes.asciidoc#L201)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

> ```
> public class CompactApplication extends Div {
> ```
This is a bad example because it is kind of impossible - both the `@Theme` annotation and the `@HtmlImport` should be on a root layout, so mayhaps you should change it to:

```
@HtmlImport("frontend://bower_components/vaadin-lumo-styles/presets/compact.html")
@Theme(value = Lumo.class)
public class MainLayout extends Div implements RouterLayout {
}
```

</blockquote></details>

Fixed. Used `AppMainLayout` for the name as `MainLayout` was already reserved in `ReadyMadeThemes.java`


<!-- Sent from Reviewable.io -->

tomivirkki commented:
___
*[documentation/theme/using-component-themes.asciidoc, line 201 at r1](https://reviewable.io/reviews/vaadin/flow-and-components-documentation/442#-LWycqIu2me3-hkuPLct:-LWysb4D7NB0h3pfDb4I:b-bdohj2) ([raw file](https://github.com/vaadin/flow-and-components-documentation/blob/8ad19ae81d04ff643ac180eebc11198b43ebd879/documentation/theme/using-component-themes.asciidoc#L201)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Fixed. Used `AppMainLayout` for the name as `MainLayout` was already reserved in `ReadyMadeThemes.java`
</blockquote></details>

...changed to `CompactMainLayout` which makes sense in this context


<!-- Sent from Reviewable.io -->
